src/tox/venv.py

```diff
@@ -515,6 +515,11 @@ def _pcall(
         env = self._get_os_environ(is_test_command=is_test_command)
         bin_dir = str(self.envconfig.envbindir)
         env["PATH"] = os.pathsep.join([bin_dir, os.environ["PATH"]])
+        if "PATH" in self.envconfig.setenv:
+            env["PATH"] = os.pathsep.join([env["PATH"], self.envconfig.setenv["PATH"]])
+            # Remove redundantly duplicated directories in PATH
```

Comment:
I like the idea, but you should make sure you don't re-order path values (use a set and a list, to keep the first appearance) - also this logic should run always not just when setenv is set